File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums)
        middle = 0
        mini = float('inf')
        
        while (left != right):
            middle = (int) ((left + right)/2)
            if nums[middle] > nums[left]:
                left = middle;
                mini = min(mini, nums[middle])
            else:
                mini = min(mini, nums[middle])
                right = middle

        return min(nums[0], mini)
                

File: Data Structures & Algorithms/remove-element/submission-1.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int padding = 0;
        for (int i = nums.length - 1; i > -1; i--) {
            if (nums[i] == val) {
                int temp = nums[nums.length - 1 - padding];
                nums[nums.length - 1 - padding] = val;
                nums[i] = temp;
                padding++;
            }
        }
        return nums.length - padding;
    }
}